From 71b2041cb08a2868418f20ada4b4830f72ef437b Mon Sep 17 00:00:00 2001
From: Alexandre Frade <admfrade@gmail.com>
Date: Sun, 13 Oct 2019 03:10:39 -0300
Subject: [PATCH 2/8] XANMOD: kconfig: set PREEMPT and RCU_BOOST without delay by
 default

Signed-off-by: Alexandre Frade <admfrade@gmail.com>
---
 kernel/rcu/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

#diff --git a/kernel/Kconfig.preempt b/kernel/Kconfig.preempt
#index 1111111..1111111 100644
#--- a/kernel/Kconfig.preempt
#+++ b/kernel/Kconfig.preempt
#@@ -13,7 +13,7 @@ config PREEMPT_BUILD
# 
# choice
# 	prompt "Preemption Model"
#-	default PREEMPT_NONE
#+	default PREEMPT
# 
# config PREEMPT_NONE
# 	bool "No Forced Preemption (Server)"
diff --git a/kernel/rcu/Kconfig b/kernel/rcu/Kconfig
index 1111111..1111111 100644
--- a/kernel/rcu/Kconfig
+++ b/kernel/rcu/Kconfig
@@ -196,8 +196,8 @@ config RCU_FANOUT_LEAF
 
 config RCU_BOOST
 	bool "Enable RCU priority boosting"
-	depends on (RT_MUTEXES && PREEMPT_RCU && RCU_EXPERT) || PREEMPT_RT
-	default y if PREEMPT_RT
+	depends on (RT_MUTEXES && PREEMPT_RCU) || PREEMPT_RT
+	default y
 	help
 	  This option boosts the priority of preempted RCU readers that
 	  block the current preemptible RCU grace period for too long.
@@ -211,7 +211,7 @@ config RCU_BOOST_DELAY
 	int "Milliseconds to delay boosting after RCU grace-period start"
 	range 0 3000
 	depends on RCU_BOOST
-	default 500
+	default 0
 	help
 	  This option specifies the time to wait after the beginning of
 	  a given grace period before priority-boosting preempted RCU
